contracts/tasks/e2e.ts

```diff
@@ -23,102 +23,145 @@ task("e2e", "Runs all e2e tests")
 
     const testResults: Record<string, string> = {};
 
-    let result = await runOpenAi(
-      contractAddress,
-      "gpt-4-turbo-preview",
-      "Who is the president of USA?",
-      hre,
-    )
-    testResults["OpenAI gpt-4-turbo-preview"] = result.error || "✅";
-  
-    result = await runOpenAi(
-      contractAddress,
-      "gpt-3.5-turbo-1106",
-      "Who is the president of USA?",
-      hre,
-    )
-    testResults["OpenAI gpt-3.5-turbo-1106"] = result.error || "✅";
-  
-    result = await runOpenAiVision(
-      contractAddress,
-      "gpt-4-turbo",
-      "What is on this image",
-      "https://picsum.photos/200/300",
-      hre,
-    )
-    testResults["OpenAI gpt-4-turbo vision"] = result.error || "✅";
-  
-    result = await runOpenAiVision(
-      contractAddress,
-      "gpt-4o",
-      "What is on this image",
-      "https://picsum.photos/200/300",
-      hre,
-    )
-    testResults["OpenAI gpt-4o"] = result.error || "✅";
+    try {
```

Comment:
hmm why are these try catches needed? The runOpenAi already has it as well?

Reply:
was about to write a comprehensive response :D